### PR TITLE
Allow logout and make surveys optional for user admin

### PR DESCRIPTION
**Pull request by Muddasser**

### CODE REVIEW

File: lms/djangoapps/onboarding_survey/forms.py [hunk 1]
```diff
@@ -200,5 +199,8 @@ def clean(self):
 
         year = cleaned_data['founding_year']
 
-        if len("{}".format(year)) < 4 or year < 0:
-            raise forms.ValidationError("You entered an invalid year format. Please enter a valid year with 4 digits.")
+        if year:
```
Comment (shahbazn) [inline lms/djangoapps/onboarding_survey/forms.py]: is year optional?

File: lms/djangoapps/onboarding_survey/middleware.py [hunk 2]
```diff
@@ -22,12 +23,15 @@ class RedirectMiddleware(object):
...
     def process_request(self, request):
 
-        if request.is_ajax():
+        if request.is_ajax() or request.get_full_path() == '/logout':
             return None
 
         if not request.user.is_anonymous():
             user = request.user
 
+            if user.username == settings.ADMIN_USER_NAME:
```
Comment (shahbazn) [inline lms/djangoapps/onboarding_survey/middleware.py]: what if we have multiple admin users? please get admin info from user object. user.is_superuser

File: lms/envs/common.py [hunk 3]
```diff
@@ -3040,3 +3040,5 @@
 ############## Settings for the Enterprise App ######################
 
 ENTERPRISE_ENROLLMENT_API_URL = LMS_ROOT_URL + "/api/enrollment/v1/"
+
```
Comment (shahbazn) [inline lms/envs/common.py]: dont need this

File: lms/djangoapps/onboarding_survey/forms.py (hunk 1 above)
Comment (Muddasser) [reply lms/djangoapps/onboarding_survey/forms.py]: That's right.

**Pull request merged**